opencti-platform/opencti-front/src/private/components/common/lists/FiltersElement.js

```diff
@@ -0,0 +1,157 @@
+import Grid from '@mui/material/Grid';
+import TextField from '@mui/material/TextField';
+import * as R from 'ramda';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import Autocomplete from '@mui/material/Autocomplete';
+import React from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import ItemIcon from '../../../../components/ItemIcon';
+import { useFormatter } from '../../../../components/i18n';
+import { directFilters, FiltersVariant } from '../../../../utils/filters/filtersUtils';
+
+const useStyles = makeStyles((theme) => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+    color: theme.palette.primary.main,
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+}));
+
+const FiltersElement = ({
+  variant,
+  keyword,
+  availableFilterKeys,
+  handleChangeKeyword,
+  handleChangeDate,
+  handleAcceptDate,
+  handleValidateDate,
+  noDirectFilters,
+  inputValues,
+  searchScope,
+  entities,
+  handleChange,
+  searchEntities,
+  renderSearchScopeSelection,
+}) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+
+  return (
+    <Grid container={true} spacing={2}>
+      {variant === FiltersVariant.dialog && (
+        <Grid item={true} xs={12}>
+          <TextField
+            label={t('Global keyword')}
+            variant="outlined"
+            size="small"
+            fullWidth={true}
+            value={keyword}
+            onChange={handleChangeKeyword}
+          />
+        </Grid>
+      )}
+      {R.filter(
+        (n) => noDirectFilters || !R.includes(n, directFilters),
+        availableFilterKeys,
+      ).map((filterKey) => {
+        if (
+          filterKey.endsWith('start_date')
+          || filterKey.endsWith('end_date')
+        ) {
+          return (
+            <Grid key={filterKey} item={true} xs={6}>
+              <DatePicker
+                label={t(`filter_${filterKey}`)}
+                value={inputValues[filterKey] || null}
+                variant="inline"
+                disableToolbar={false}
+                autoOk={true}
+                allowKeyboardControl={true}
+                onChange={(value) => handleChangeDate(filterKey, value)}
+                onAccept={(value) => handleAcceptDate(filterKey, value)}
+                renderInput={(params) => (
+                  <TextField
+                    variant="outlined"
+                    size="small"
+                    fullWidth={variant === 'dialog'}
+                    onKeyDown={(event) => handleValidateDate(filterKey, event)}
+                    {...params}
+                  />
+                )}
+              />
+            </Grid>
+          );
+        }
+        let options = [];
+        if (['fromId', 'toId'].includes(filterKey)) {
```

Comment:
Not sure how we could make it more readable with a function ?

Reply:
I mean, to don't have it directly in the html part.

Reply:
Oh, yes we definitely should. For this task I choose not to update the way we do things because there is a lot to do and it was a side task from the OpenVocab one. But you're right, we should someday rewrite the Filters code to make it more friendly :)